packages/mds-audit/api.ts

```diff
@@ -153,81 +154,85 @@ function api(app: express.Express): express.Express {
    * initiate an audit
    * @param {UUID} audit_trip_id unique ID of this audit record (client-generated)
    */
-  app.post(pathsFor('/trips/:audit_trip_id/start'), async (req: AuditApiAuditStartRequest, res: AuditApiResponse) => {
-    try {
-      const { audit_trip_id, audit, audit_subject_id, recorded } = res.locals
-
-      if (!audit) {
-        const {
-          timestamp,
-          provider_id,
-          provider_vehicle_id,
-          audit_event_id = uuid(),
-          audit_device_id,
-          telemetry
-        } = req.body
-
-        // Validate input params
-        if (
-          isValidTimestamp(timestamp) &&
-          isValidProviderId(provider_id) &&
-          isValidProviderVehicleId(provider_vehicle_id) &&
-          isValidAuditEventId(audit_event_id) &&
-          isValidAuditDeviceId(audit_device_id) &&
-          isValidTelemetry(telemetry, { required: false })
-        ) {
-          // Find provider device by vehicle id lookup
-          const provider_device = await readDeviceByVehicleId(provider_id, provider_vehicle_id)
-          const provider_device_id = provider_device ? provider_device.device_id : null
-          const provider_name = providerName(provider_id)
-
-          // Create the audit
-          await writeAudit({
-            audit_trip_id,
-            audit_device_id,
-            audit_subject_id,
-            provider_id,
-            provider_name,
-            provider_vehicle_id,
-            provider_device_id,
-            timestamp,
-            recorded
-          })
-
-          // Create the audit start event
-          await writeAuditEvent({
-            audit_trip_id,
-            audit_event_id,
-            audit_subject_id,
-            audit_event_type: AUDIT_EVENT_TYPES.start,
-            ...flattenTelemetry(telemetry),
-            timestamp,
-            recorded
-          })
+  app.post(
```

Comment:
Sorry, I'm probably just bad at reading the Express documentation, but does each of these `app.BLAH` methods take an arbitrary number of arguments? Does it call each of them one after another?

Reply:
See `Route Handlers` on https://expressjs.com/en/guide/routing.html
```
var cb0 = function (req, res, next) {
  console.log('CB0')
  next()
}

var cb1 = function (req, res, next) {
  console.log('CB1')
  next()
}

var cb2 = function (req, res) {
  res.send('Hello from C!')
}

app.get('/example/c', [cb0, cb1, cb2])
```

Reply:
Also, the array syntax is optional. `app.get('/example/c', cb0, cb1, cb2)` or `app.get('/example/c/', [cb0, cb1], cb2)` all work identically. The key differentiator is that the inner functions are middleware functions that need to call `next()` in order to pass control to subsequent middleware functions and ultimately the route handler.

Reply:
Ah thank you, that is very enlightening.